src/connection.js

```diff
@@ -567,7 +570,8 @@ class Connection extends EventEmitter {
 
   clearRequestTimer() {
     if (this.requestTimer) {
-      return clearTimeout(this.requestTimer);
+      clearTimeout(this.requestTimer);
+      this.requestTimer = undefined;
```

Comment:
Right, that is the behavior I was expecting with the current test and we should keep it.

My suggestion is to add a new test like this.

- Send any query that returns multiple rows.
- In the first invocation of 'row' event handler, sleep for a little over the request timeout period.
   o Something like child_process.execSync('sleep 2')

In the old code this should cause the request to timeout. With your fix, this should work because you clear the request timer already. Does that make sense?